### PR TITLE
Bootstrap 3 Modal Pane

### PR DESCRIPTION
**Pull request by boomsb**
Added modal-dialog and modal-content wrapper divs.
Condensed animation in/out to a fade property which uses bootstrap fading.
Updated tests and documentation.

### CODE REVIEW

**Comment by bradleypriest** [comment]
Does Bootstrap fade via CSS or JS?

**Comment by bradleypriest** [comment]
The reason I ask is because from a cursory glance it looks like your relying on the Bootstrap JS, which ember-bootstrap doesn't assume. I may be mistaken though

**Comment by boomsb** [comment]
I believe it's using CSS transitions triggered in JS, and I coded it with a dependency on bootstrap.js. I didn't realize that would be an issue. I'll see what I can do about using straight jQuery and modify the CSS manually.

Also, anyone know why this.$() would be failing in the tests? It appears to be a valid Ember.View function for getting the current jQuery element.

**Comment by boomsb** [comment]
I've removed the bootstrap.js dependency and got all the updated and passing again.

File: packages/ember-bootstrap/lib/views/modal_pane.js [hunk 1]
```diff
@@ -40,24 +45,28 @@ Bootstrap.ModalPane = Ember.View.extend(Ember.DeferredMixin, {
   }),
 
   didInsertElement: function() {
-    if (get(this, 'showBackdrop')) this._appendBackdrop();
+    if (get(this, 'showBackdrop'))
```
Comment (bradleypriest) [inline packages/ember-bootstrap/lib/views/modal_pane.js]: IMHO if we're going to make code style changes, we should at least try and head towards ember-core style. i.e. curly braces when making if statements.

File: packages/ember-bootstrap/lib/views/modal_pane.js [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+    '{{#if view.parentView.secondary}}<button type="button" class="btn btn-default" rel="secondary">{{view.parentView.secondary}}</button>{{/if}}',
+    '{{#if view.parentView.primary}}<button type="button" class="btn btn-primary" rel="primary">{{view.parentView.primary}}</button>{{/if}}' ]
+    .join("\n");
 
-var modalPaneBackdrop = '<div class="modal-backdrop"></div>';
+var modalPaneBackdrop = '<div class="modal-backdrop in"></div>';
 
 Bootstrap.ModalPane = Ember.View.extend(Ember.DeferredMixin, {
-  classNames: 'modal',
+  classNames: [ 'modal', 'in' ],
```
Comment (dmathieu) [inline packages/ember-bootstrap/lib/views/modal_pane.js:25]: Why have you added this in class? (same question for two lines above).

**Comment by boomsb** [comment]
On the background it sets the opacity and alpha values.

On the modal it's not doing much. I believe it's used for some CSS transforms if you add the fade class, probably not necessary, but not harmful either.

**Comment by analytically** [comment]
+1

**Pull request merged**